synapse/storage/databases/main/roommember.py

```diff
@@ -396,6 +397,7 @@ def _get_room_summary_txn(
         )
 
     @cached()
+    @trace
```

Comment:
It looks like this `@trace` call is messing up some of the tests. The method still gives results so it's unclear to me why it would change things 🤔

https://github.com/matrix-org/synapse/runs/7891614050?check_suite_focus=true#step:7:3384
```
[FAIL]
Traceback (most recent call last):
  File "/home/runner/work/synapse/synapse/tests/rest/client/test_rooms.py", line 726, in test_post_room_initial_state
    self.assertEqual(50, channel.resource_usage.db_txn_count)
  File "/home/runner/.cache/pypoetry/virtualenvs/matrix-synapse-pswDeSvb-py3.9/lib/python3.9/site-packages/twisted/trial/_synctest.py", line 422, in assertEqual
    super().assertEqual(first, second, msg)
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/unittest/case.py", line 837, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/opt/hostedtoolcache/Python/3.9.13/x64/lib/python3.9/unittest/case.py", line 830, in _baseAssertEqual
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: 50 != 49

tests.rest.client.test_rooms.RoomsCreateTestCase.test_post_room_initial_state
```

Reply:
Wonder if it's because the function being returned by `trace` always has the same `__name__` and so just a guess, but these two get the same cache entries. Seems like a massive trap. I'll see if I can confirm.
```python3
@cached()
@trace
def fun1(): ...

@cached()
@trace
def fun2(): ...
```

Reply:
I'll re-introduce this in another PR if I get an inkling to have it again.